### PR TITLE
Feature/stairs on dom

### PR DESCRIPTION
**Pull request by leighlars**
### Type of change made:
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features
- [X] Refactoring
- [ ] Planning

### Detailed Description
I was able to clear the input from hydration and stairs after submission.
The user can now interact with the stairs card by adding in their steps for the day, viewing the flights and stairs records, and viewing the community average for flights per day.
I wrote tests for the stairs portion of tonight's work, and modified some  methods/tests in Hydro based on the excellent work Steph did in getting the week average methods. 
I see some opportunities for refactoring in steps, and probably stairs. Wanna ask Steph her thoughts (I added notes)
The DOM updates file is looking cleaner by the day!!!
The leaderBoard now renders but the CSS isn't working properly.

### Why is this change required? What problem does it solve?
See above. 

### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
Not particularly. BeforeEach gave me trouble but I basically had to mock the entire start up of the app and create a fake rawData array. 

### Files modified:
- [X] index.html
- [ ] .scss // file name(s):
- [X] classes // file name(s): Hydro, Activity, User Repo
- [X] test // files name(s): Hydro, Activity, User Repo
- [X] scripts.js
- [X] Other files: DOM Updates

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -28,31 +28,29 @@ const domUpdates = {
...
+        paragraph.classList.add('green-text');
+      }
+      if (friendsStepsParagraphs[friendsStepsParagraphs.length - 1] === paragraph) {
+        paragraph.classList.add('red-text');
+      }
+      if (paragraph.innerText.includes('YOU')) {
+        paragraph.classList.add('yellow-text');
+      }
+    });
+  },
```
Comment (NakiNorton) [inline src/domUpdates.js:53]: Great work getting this cleaned up.

File: src/User.js [hunk 2]
```diff
@@ -6,97 +6,73 @@ class User {
   constructor(userDetails, todayDate) {
     this.id = this.checkUserId(userDetails.id);
     this.name = this.checkName(userDetails.name);
-    this.address = userDetails.address || 'No address added.';
-    this.email = userDetails.email || 'No email address added.';
-    this.strideLength = userDetails.strideLength || 'Stride length not added.';
-    this.dailyStepGoal = userDetails.dailyStepGoal || 'Daily step goal not added.';
-    this.friends = userDetails.friends || 'Add friends for friendly competition!';
+    this.address = userDetails.address || "No address added.";
+    this.email = userDetails.email || "No email address added.";
```
Comment (Leasw144) [inline src/User.js:10]: Good job keeping the quotes consistent

File: src/domUpdates.js [hunk 3]
```diff
@@ -28,31 +28,29 @@ const domUpdates = {
...
-  //     }
-  //     if (paragraph.innerText.includes('YOU')) {
-  //       paragraph.classList.add('yellow-text');
-  //     }
-  //   });
-  // }
+    this.showLeaderBoard(); // CSS is finnicky on it
+  },
+
+  showLeaderBoard() {
```
Comment (Leasw144) [inline src/domUpdates.js:34]: This looks a lot nicer than what it was before

**Comment by NakiNorton** [comment]
Thanks for all your work on this! You covered a lot of ground. There's logic in here that should be applied to other sections of our project, namely - the clear input functionality in domUpdates, and the addInput method.

**Comment by Leasw144** [comment]
The amount of code you wrote in one night is jaw dropping. I'll merge and incorporate this into what I'm doing now.

**Pull request merged**